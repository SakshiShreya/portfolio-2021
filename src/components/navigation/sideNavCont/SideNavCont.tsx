import { useContext } from "react";
import { Link } from "react-router-dom";
import { DeviceContext } from "../../../context/DeviceContext";
import { NavigationContext } from "../../../context/NavigationContext";
import styles from "./SideNavCont.module.scss";

const links = [
  { title: "Home", link: "/" },
  { title: "About Me", link: "/about" },
  { title: "Experience", link: "/experience" },
  { title: "Contact me", link: "/contact" }
];

export interface SideNavContProps {}

const SideNavCont: React.FC<SideNavContProps> = props => {
  const { isOpen, onOpen } = useContext(NavigationContext);
  const { device } = useContext(DeviceContext);

  return (
    <nav className={[styles.drawer, isOpen ? styles.show : null].join(" ")}>
      {device === "desktop" && <img src={process.env.PUBLIC_URL + "/assets/logo/SSlogo.png"} className={styles.logo} alt="Logo" />}
      <ul>
        {links.map(link => (
          <li className={styles.li + " h4"} key={link.title}>
            <Link to={link.link} onClick={() => device !== "desktop" && onOpen(!isOpen)}>
              {link.title}
            </Link>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default SideNavCont;
