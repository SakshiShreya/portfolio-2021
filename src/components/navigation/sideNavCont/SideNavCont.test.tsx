import { HashRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import SideNavCont from "./SideNavCont";

it("should render SideNavCont properly", () => {
  const elem = renderer
    .create(
      <HashRouter>
        <SideNavCont />
      </HashRouter>
    )
    .toJSON();
  expect(elem).toMatchSnapshot();
});
