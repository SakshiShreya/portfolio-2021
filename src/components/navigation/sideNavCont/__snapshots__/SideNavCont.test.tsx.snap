// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SideNavCont properly 1`] = `
<nav
  className="drawer "
>
  <img
    alt="Logo"
    className="logo"
    src="/assets/logo/SSlogo.png"
  />
  <ul>
    <li
      className="li h4"
    >
      <a
        href="#/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="li h4"
    >
      <a
        href="#/about"
        onClick={[Function]}
      >
        About Me
      </a>
    </li>
    <li
      className="li h4"
    >
      <a
        href="#/experience"
        onClick={[Function]}
      >
        Experience
      </a>
    </li>
    <li
      className="li h4"
    >
      <a
        href="#/contact"
        onClick={[Function]}
      >
        Contact me
      </a>
    </li>
  </ul>
</nav>
`;
